src/Nethermind/Nethermind.Shutter/IShutterBlockHandler.cs

```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Core;
+
+namespace Nethermind.Shutter;
+public interface IShutterBlockHandler : IDisposable
+{
+    void OnNewHeadBlock(Block head);
```

Comment:
Why not pass BlockTree to constructor and subscribe there? Less things in the interface